include/swift/AST/DiagnosticsSema.def

```diff
@@ -3264,6 +3264,14 @@ ERROR(autodiff_attr_result_not_differentiable,none,
       "'Differentiable', but %0 does not conform to 'Differentiable'", (Type))
 ERROR(autodiff_attr_opaque_result_type_unsupported,none,
       "cannot differentiate functions returning opaque result types", ())
+ERROR(autodiff_attr_static_decl_mismatch,none,
+      "function must match usage of static declaration modifier "
```

Comment:
How about 
```suggestion
      "derivative function %0 operates on {an instance|a type}, not {an instance|a type} as required"
```

Reply:
You can point back at the original function with a note.

Reply:
Thank you for the suggestion. I changed the working of the error diagnostic and adapted the subsequent notes accordingly in cfd7932a507206c8354eda4405a0ce46c7fbafe3